library/alloc/src/collections/vec_deque/mod.rs

```diff
@@ -124,18 +105,16 @@ impl<T: Clone, A: Allocator + Clone> Clone for VecDeque<T, A> {
     }
 
     fn clone_from(&mut self, other: &Self) {
-        self.truncate(other.len());
-
-        let mut iter = PairSlices::from(self, other);
-        while let Some((dst, src)) = iter.next() {
-            dst.clone_from_slice(&src);
-        }
+        self.clear();
+        self.head = 0;
+        self.reserve(other.len);
 
-        if iter.has_remainder() {
-            for remainder in iter.remainder() {
-                self.extend(remainder.iter().cloned());
-            }
+        let (a, b) = other.as_slices();
+        unsafe {
```

Comment:
Please *always* have `// SAFETY` comments on `unsafe` blocks saying why the safety preconditions of the called things are met.

But for this specific case, is `unsafe` even needed?  Is `self.extend(other.iter().cloned())` not going to do the right thing here?  So long as that uses internal iteration to consume the iterator, your `vec_deque::Iter`'s `fold` method will do the correct loop splitting that should make it work great.